### PR TITLE
[DOCS] Updating Contribute to OpenVINO articles

### PR DESCRIPTION
**Pull request by sgolebiewski-intel**
Updating the existing articles about making contributions to OpenVINO, adding an article to docs.openvino.ai.

### CODE REVIEW

File: CONTRIBUTING_DOCS.md [hunk 1]
```diff
@@ -2,110 +2,709 @@
...
+    | command-line    | command line   |                                                                                   |
+    | device-specific |                |                                                                                   |
+    | end-to-end      |                | end to end may be used as an adjective                                            |
+    | frontend        | frontend       |  never front-end                                                                  |
+    | open-source     | open source    | never open-sourced (even with the meaning of 'released as open source')           |
+    | pre-processing  | pre-processing | due to the popularity criterion                                                   |
+    | pre-release     |                | the noun "prerelease" should be avoided.                                          |
+    | pre-trained     |                |  should be avoided in meanings other than the adjectival. click here for details. |
+    | pre-trained     |                |  should be avoided in meanings other than the adjectival. click here for details. |
```
[the remark this answers is not in the document]
Comment (sgolebiewski-intel) [reply CONTRIBUTING_DOCS.md:567]:
```suggestion
```

File: CONTRIBUTING_DOCS.md [hunk 2]
```diff
@@ -2,110 +2,709 @@
 
 ## Basic article structure
 
-OpenVINO documentation is built using Sphinx and the reStructuredText formatting. 
+OpenVINO documentation is built using Sphinx and the reStructuredText formatting.
```
[the remark this answers is not in the document]
Comment (sgolebiewski-intel) [reply CONTRIBUTING_DOCS.md:5]:
```suggestion
OpenVINO documentation is built using [Sphinx](https://www.sphinx-doc.org/) and the [reStructuredText](https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html) formatting.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,155 +1,162 @@
...
+it addresses.
 
 ### 5. Wait for a review
 
-We'll make sure to review your Pull Request as soon as possible and provide you with our feedback. You can expect a merge once your changes are validated with automatic tests and approved by maintainers.
+We will make sure to review your **Pull Request** as soon as possible and provide feedback.
+You can expect a merge once your changes have been validated with automatic tests and
+approved by maintainers.
 
-## License
```
[the remark this answers is not in the document]
Comment (sgolebiewski-intel) [reply CONTRIBUTING.md:152]: I have restored the note about LICENSE in https://github.com/openvinotoolkit/openvino/pull/23650/commits/ed4b603c04eb97a3f1ddd4b2d44b92da36bd760a

File: CONTRIBUTING_DOCS.md [hunk 4]
```diff
@@ -2,110 +2,709 @@
...
 That means the basic formatting rules need to be used:
```
[the remark this answers is not in the document]
Comment (sgolebiewski-intel) [reply CONTRIBUTING_DOCS.md:31]: I have applied the TOC in https://github.com/openvinotoolkit/openvino/pull/23650/commits/ed4b603c04eb97a3f1ddd4b2d44b92da36bd760a

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,155 +1,162 @@
...
+     The documentation is built from several sources, mainly
+     [docs](./docs) folder, using [Sphinx](https://www.sphinx-doc.org/) and the
+     [reStructuredText](https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html)
+     markup language. It is published at [docs.openvino.ai](https://docs.openvino.ai/),
+     which is recommended for reading the articles.
 
-   * **User documentation** is built from several sources and published at 
-     [docs.openvino.ai](https://docs.openvino.ai/), which is the recommended place for reading 
-     these documents. Use the files maintained in this repository only for editing purposes. 
+   * **Check OpenVINO developer documentation**
```
[the remark this answers is not in the document]
Comment (sgolebiewski-intel) [reply CONTRIBUTING.md:71]: How about:

```suggestion
   * **Check OpenVINO developer documentation**
     
     The intended audience for the developer documentation is developers. This documentation
     includes detailed information on the architecture of OpenVINO. It helps developers understand
     the process of building the toolkit, so they can modify and implement it as per the requirement.
```

File: docs/articles_en/about-openvino/contributing.rst [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+   contributing/code-contribution-guide
+
+OpenVINO™ is always looking for opportunities to improve and your contributions
+play a big role in this process. Here are four ways you can make OpenVINO better:
+
+- `Provide feedback <#provide-feedback>`__
+- `Contribute code changes <#contribute-code-changes>`__
+- `Improve documentation <#improve-documentation>`__
+- `Promote and support OpenVINO <#promote-and-support-openvino>`__
+
```
Comment (p-wysocki) [inline docs/articles_en/about-openvino/contributing.rst:17]: - Pick up a Good First Issue?

File: docs/articles_en/about-openvino/contributing.rst [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+- Open a new `GitHub Discussion <https://github.com/openvinotoolkit/openvino/discussions>`__.
+- Create a `Feature Request Issue <https://github.com/openvinotoolkit/openvino/issues/new?assignees=octocat&labels=enhancement%2Cfeature&projects=&template=feature_request.yml&title=%5BFeature+Request%5D%3A+>`__
+  if your idea is already well defined.
+
+In both cases, provide a detailed description and list potential use cases,
+benefits, and challenges. Keep in mind that even if your input is not immediately
+prioritized, it may be used at a later or undertaken by the community.
+
+
+:fas:`code-branch` Contribute code changes
```
Comment (p-wysocki) [inline docs/articles_en/about-openvino/contributing.rst:43]: GFI could also be mentioned here

File: docs/articles_en/about-openvino/contributing.rst [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+:fas:`bullhorn` Promote and support OpenVINO
+############################################
+
+.. rubric:: Popularize OpenVINO
+   :name: popularize-openvino
+
+Articles, tutorials, blog posts, demos, videos, and any other involvement in the
+OpenVINO community is more than welcome. If you discuss or present OpenVINO on
+various social platforms, you are raising awareness of the product among A.I.
```
Comment (p-wysocki) [inline docs/articles_en/about-openvino/contributing.rst:138]:
```suggestion
various social platforms, you are raising awareness of the product among AI
```

File: docs/articles_en/about-openvino/contributing.rst [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+.. note::
+
+   By contributing to the OpenVINO project, you agree that your contributions
+   will be licensed under `the terms of the OpenVINO repository <https://github.com/openvinotoolkit/openvino/blob/master/LICENSE>`__.
+
+
+Additional Resources
+#####################
+
```
Comment (p-wysocki) [inline docs/articles_en/about-openvino/contributing.rst:163]: https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md#Technical-guide could also be added.

File: docs/articles_en/about-openvino/contributing/code-contribution-guide.rst [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+   To get assigned to a task, simply leave a comment with the ``.take`` command in
+   the selected issue. You can always ask OpenVINO developers for guidance,
+   both technical and organizational:
+
+   - assign users in the **“Contact points”** section,
+   - visit `Intel DevHub Discord server <https://discord.gg/7pVRxUwdWG>`__ to ask
+     questions in the channel dedicated to **“Good First Issue”** support, or any other.
+
+3. **Build OpenVINO.**
```
Comment (p-wysocki) [inline docs/articles_en/about-openvino/contributing/code-contribution-guide.rst:61]: This point could be renamed to "Follow technical contributing guide" and point to https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md#Technical-guide.

File: docs/articles_en/about-openvino.rst [hunk 11]
```diff
@@ -11,6 +9,7 @@ About OpenVINO
    about-openvino/performance-benchmarks
    about-openvino/compatibility-and-support
    Release Notes <about-openvino/release-notes-openvino>
+   about-openvino/contributing
```
Comment (sgolebiewski-intel) [inline docs/articles_en/about-openvino.rst:12]:
```suggestion
   about-openvino/contributing
   Release Notes <about-openvino/release-notes-openvino>
```

File: docs/articles_en/about-openvino/contributing.rst (hunk 9 above)
Comment (sgolebiewski-intel) [reply docs/articles_en/about-openvino/contributing.rst:163]: > https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md#Technical-guide could also be added.

The article below, Code Contribution Guide is based on the technical guide you mention :)

File: docs/articles_en/about-openvino/contributing/code-contribution-guide.rst (hunk 10 above)
Comment (sgolebiewski-intel) [reply docs/articles_en/about-openvino/contributing/code-contribution-guide.rst:61]: This article is based on the "technical guide" and reference to the source article in this case is not needed.

File: docs/articles_en/about-openvino/contributing.rst (hunk 6 above)
Comment (sgolebiewski-intel) [reply docs/articles_en/about-openvino/contributing.rst:17]: It is strongly highlighted in "Code Contribution Guide".

File: docs/articles_en/about-openvino/contributing.rst (hunk 7 above)
Comment (sgolebiewski-intel) [reply docs/articles_en/about-openvino/contributing.rst:43]: GFI is emphasized in the tip admonition below, line 60.

Comment (sgolebiewski-intel) [reply docs/articles_en/about-openvino/contributing.rst:43]: image

**Review by p-wysocki — APPROVED** [review]

**Review by jacekpawlak — APPROVED** [review]

**Pull request merged**